packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksEditor.tsx

```diff
@@ -18,8 +25,52 @@ import { withStrapiSchema } from './plugins/withStrapiSchema';
  * BlocksEditorProvider
  * -----------------------------------------------------------------------------------------------*/
 
+interface NonSelectorBlock {
+  renderElement: (props: RenderElementProps) => React.JSX.Element;
+  value: object;
+  matchNode: (node: Attribute.BlocksNode) => boolean;
+  isInBlocksSelector: false;
+  handleEnterKey?: (editor: Editor) => void;
+  handleBackspaceKey?: (editor: Editor, event: React.KeyboardEvent<HTMLElement>) => void;
+}
+
+type SelectorBlock = Omit<NonSelectorBlock, 'isInBlocksSelector'> & {
+  isInBlocksSelector: true;
+  icon: React.ComponentType;
+  label: MessageDescriptor;
+};
+
+type NonSelectorBlockKey = 'list-item' | 'link';
+
+const selectorBlockKeys = [
+  'paragraph',
+  'heading-one',
+  'heading-two',
+  'heading-three',
+  'heading-four',
+  'heading-five',
+  'heading-six',
+  'list-ordered',
+  'list-unordered',
+  'image',
+  'quote',
+  'code',
+] as const;
+
+type SelectorBlockKey = (typeof selectorBlockKeys)[number];
+
+const isSelectorBlockKey = (key: unknown): key is SelectorBlockKey => {
+  return typeof key === 'string' && selectorBlockKeys.includes(key as SelectorBlockKey);
+};
```

Comment:
Not related to your PR:

Is this function even necessary I only see it used in the BlocksToolbar but couldn't we just type parameter instead of using unknown? Don't we know 🤔 

```diff
- const selectOption = (optionKey: unknown) => {
-  if (!isSelectorBlockKey(optionKey)) {
-    return;
+ const selectOption = (optionKey: SelectorBlockKey) => {
+  if (['list-ordered', 'list-unordered'].includes(optionKey)) {
```

Reply:
I added both this and moving withImages to the "miscellaneous improvements", Madhuri or I will look into it later this sprint 👍

Reply:
Hey @markkaylor, I looked at it again. The issue is that selectOption is passed to SingleSelect's onChange, and the expected type for onChange is `(value: string | number) => void`. I don't think I can let it know that all the options are of type SelectorBlockKey, so I'll have to keep using the type guard